consensus/src/consensus/factory.rs

```diff
@@ -192,6 +195,15 @@ impl ConsensusFactory for Factory {
                 entry
             }
         };
+
+        #[cfg(feature = "developer-mode")]
+        if is_new_consensus {
+            set_genesis_utxo_commitment_from_config(&mut config);
```

Comment:
I think you need to call this also if `is_new_consensus == false`. Otherwise genesis hash will be wrong if the node restarted